src/test/java/com/aws/greengrass/util/platforms/windows/WindowsPlatformTest.java

```diff
@@ -221,4 +243,144 @@ void GIVEN_a_well_known_group_sid_WHEN_lookupGroupByIdentifier_THEN_succeed() {
+
+        // Create a named pipe, passing null as lpSecurityAttributes (sane as Device SDK)
+        String namedPipe = windowsPlatform.prepareIpcFilepath(tempDir);
+        WinNT.HANDLE handle = Kernel32.INSTANCE.CreateNamedPipe(namedPipe,  // lpName
+                PIPE_ACCESS_DUPLEX | FILE_FLAG_OVERLAPPED,  // dwOpenMode
+                PIPE_TYPE_BYTE | PIPE_READMODE_BYTE | PIPE_WAIT | PIPE_ACCEPT_REMOTE_CLIENTS, // dwPipeMode
+                PIPE_UNLIMITED_INSTANCES, // nMaxInstances
+                512, // nOutBufferSize
+                512, // nInBufferSize
+                0, // nDefaultTimeOut
+                null); // lpSecurityAttributes
+        assertFalse(WinBase.INVALID_HANDLE_VALUE.equals(handle));
+
+        // Test access right
+        int infoType = OWNER_SECURITY_INFORMATION
+                | GROUP_SECURITY_INFORMATION
+                | DACL_SECURITY_INFORMATION;
+
+        PointerByReference ppsidOwner = new PointerByReference();
+        PointerByReference ppsidGroup = new PointerByReference();
+        PointerByReference ppDacl = new PointerByReference();
+        PointerByReference ppSecurityDescriptor = new PointerByReference();
+
+        int ret = Advapi32.INSTANCE.GetSecurityInfo(handle,
+                SE_KERNEL_OBJECT,
+                infoType,
+                ppsidOwner,
+                ppsidGroup,
+                ppDacl,
+                null,
+                ppSecurityDescriptor);
+        assertEquals(0, ret);
+
+        // Change access right
+
+        // This code should be implemented in WindowsPlatform.setIpcFilePermissions.
+        // windowsPlatform.setIpcFilePermissions(tempDir);
+        ret = Advapi32.INSTANCE.SetSecurityInfo(handle,
+                SE_KERNEL_OBJECT,
+                infoType,
+                ppsidOwner.getValue(),
+                ppsidGroup.getValue(),
+
+                // https://docs.microsoft.com/en-us/windows/win32/secauthz/access-control-lists
+                // "If the object does not have a DACL, the system grants full access to everyone."
+                ppDacl.getValue(),
+
+                null);
+        assertEquals(0, ret); // ERROR_ACCESS_DENIED
+
+        // Test access right again
+        ppsidOwner = new PointerByReference();
+        ppsidGroup = new PointerByReference();
+        ppDacl = new PointerByReference();
+        ppSecurityDescriptor = new PointerByReference();
+
+        ret = Advapi32.INSTANCE.GetSecurityInfo(handle,
+                SE_KERNEL_OBJECT,
+                infoType,
+                ppsidOwner,
+                ppsidGroup,
+                ppDacl,
+                null,
+                ppSecurityDescriptor);
+        assertEquals(0, ret);
+    }
+
+    // Copied from: https://github.com/java-native-access/jna/blob/master/contrib/platform/test/com/sun/jna/platform/win32/Advapi32Test.java#L1319
+    @Test
+    public void testGetSetSecurityInfoNoSACL() throws Exception {
```

Comment:
This test is copied from the JNA project. It creates a file, calls `GetSecurityInfo` and `SetSecurityInfo` on the file handle. I run the test for sanity check and it is passing.